internal/testutils/log.go

```diff
@@ -38,12 +36,5 @@ func LogsToBuffer(level log.Level, t *testing.T) *bytes.Buffer {
 	buf := new(bytes.Buffer)
 	log.SetOutput(buf)
```

Comment:
Shouldn't the output reset to original value after ? (like a file or os.Stderr)
Wdyt of this [stack overflow](https://stackoverflow.com/a/68807754) example ?
An other way to do it would be to redirect log output to a file, and read it in one row, see for instance traefik [integration test on acccess log](https://github.com/traefik/traefik/blob/master/integration/access_log_test.go).

Reply:
Traefik integration test example will work, but it cover a case when there is a single logger. Exactly why I deleted `klog` from the intial function.

At the moment we not testing `klog` logs, for that we could have distinct function.

I agree with string, but not sure about paralel `-race` writes to file as IO is async, so not sure but we could try if problem still exist. The problelm with `bytes.Buffer` - is a slice where mutliple go routines could concurrently write to buffer

> Shouldn't the output reset to original value after ?

Yes this is another option, aka `buf.Reset()`.

Probably worth to re-run few times to validate that `-race` no longer catchin data races